### PR TITLE
Removed unused metrics

### PR DESCRIPTION
**Pull request by aarshkshah1992**
Removes metrics we no longer need as the backpressure PR has landed in main and some of those metrics were put in place to catch old bugs that no longer exist.

### CODE REVIEW

File: fetcher.go [hunk 1]
```diff
@@ -324,92 +307,20 @@ func (p *pool) fetchResource(ctx context.Context, from *Node, resource string, m
 
 	// trace-metrics
 	// request life-cycle metrics
-
-	fetchRequestSuccessTimeTraceMetric.WithLabelValues(resourceType, getCacheStatus(isCacheHit), "tcp_connection").Observe(float64(result.TCPConnection.Milliseconds()))
```
Comment (willscott) [inline fetcher.go:328]: do we not want this l1 breakdown timing anymore?

Comment (aarshkshah1992) [reply fetcher.go:328]: @willscott It hasn't helped us with anything so far. We know where most time is spent on for hits/misses and we already have these same metrics on Saturn as well. 
I think the e2e tracing work will be more helpful here.

Comment (willscott) [reply fetcher.go:328]: we want to be able to subtract it from overall request latency to separate out the L1 / caboose proportion from the lassie proportion in the long tail, right?

Comment (aarshkshah1992) [reply fetcher.go:328]: Hmm, okay, no harm in keeping it then. I've put it back.

**Comment by aarshkshah1992** [comment]
@willscott Can I merge ?

**Comment by willscott** [comment]
waiting for visibility from @AmeanAsad  when he wakes up, since it merge-conflicts with his in-progress PR to the same branch

**Pull request merged**